splunk-otel-android/src/test/java/com/splunk/rum/ActivityTracerCacheTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.rum;
+
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import android.app.Activity;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class ActivityTracerCacheTest {
+
+    @Mock Activity activity;
+
+    @Mock ActivityTracer activityTracer;
+    @Mock Function<Activity, ActivityTracer> tracerCreator;
+    AtomicReference<String> initialActivity;
+
+    @BeforeEach
+    void setup() {
+        initialActivity = new AtomicReference<>();
+    }
+
+    @Test
+    void addEventNewActivity() {
+        when(tracerCreator.apply(activity)).thenReturn(activityTracer);
+        when(activityTracer.addEvent(anyString())).thenReturn(activityTracer);
+
+        ActivityTracerCache underTest = new ActivityTracerCache(tracerCreator);
+        ActivityTracer result = underTest.addEvent(activity, "beep");
+        assertSame(activityTracer, result);
+        verify(activityTracer).addEvent("beep");
+        verifyNoMoreInteractions(tracerCreator);
+    }
+
+    @Test
+    void addEventExistingActivity() {
+        when(tracerCreator.apply(activity)).thenReturn(activityTracer);
+        when(activityTracer.addEvent(anyString())).thenReturn(activityTracer);
+
+        ActivityTracerCache underTest = new ActivityTracerCache(tracerCreator);
+        ActivityTracer result1 = underTest.addEvent(activity, "beep1");
+        ActivityTracer result2 = underTest.addEvent(activity, "beep2");
+        ActivityTracer result3 = underTest.addEvent(activity, "beep3");
+        assertSame(activityTracer, result1);
+        assertSame(activityTracer, result2);
+        assertSame(activityTracer, result3);
```

Comment:
This test will always pass even if you remove the cache, because of how the `tracerCreator` mock is set up. Consider doing 
```java
when(tracerCreator.apply(activity)).thenReturn(activityTracer, someOtherTracer);
```
and then verify that the other tracer was never returned

Reply:
I confirmed that it does still pass. :( I think the suggestion falls a little short -- the real intent is to confirm that multiple tracers weren't created by the factory, but that approach just makes sure that another one wasn't created. 

I think adding
`verify(tracerCreator).apply(activity);` at the end (with the implied `times(1)`) does what we want.